development/image-builder/main.go

```diff
@@ -198,6 +199,18 @@ func runBuildJob(o options, vs Variants, envs map[string]string) error {
 	if err != nil {
 		return err
 	}
+
+	// (@KacperMalachowski): Consider refactoring for better flexibility
+	if o.Config.PreBuildScript != "" && len(parsedTags) > 0 {
+		cmd := exec.Command(o.Config.PreBuildScript, parsedTags[len(parsedTags)-1])
+		err := cmd.Run()
+		if err != nil {
+			return fmt.Errorf("error %v", err)
+		}
```

Comment:
Arę you sure that the result of this command will be available in further commands and is not used as separate subprocess?

Reply:
It's used as subprocess and image builder only checks if command execution is successful. It fails build process on prebuild command fail.

We can consider using output in build process, but it's for now simplest solution for identified already requirement. Busola need docker-tag to update version in UI. To do that, they write it to a YAML file using Makefile, which takes it from env variable exported in build-generic.sh.